mkosi/config.py

```diff
@@ -1976,12 +1976,15 @@ def parse_config( path: Path, namespace: argparse.Namespace, defaults: argparse.
         if path.exists():
             logging.debug(f"Including configuration file {Path.cwd() / path}")
 
-            for _, k, v in parse_ini(path, only_sections=["Distribution", "Output", "Content", "Validation", "Host", "Preset"]):
+            for section, k, v in parse_ini(path, only_sections=["Distribution", "Output", "Content", "Validation", "Host", "Preset"]):
                 ns = defaults if k.startswith("@") else namespace
 
                 if not (s := settings_lookup_by_name.get(k.removeprefix("@"))):
                     die(f"Unknown setting {k}")
 
+                if section != s.section:
+                    logging.warning(f"Setting {k} should be configure in [{s.section}], not [{section}].")
```

Comment:
"should be configured"

Reply:
This wasn't fixed?

Reply:
It was, then I ran the tests with `die()` instead of `logging.warning()` to catch any misconfiguration and reverted too far before committing 🤦‍♂️.